src/compute/sort/common.rs

```diff
@@ -134,12 +123,12 @@ where
                 // Soundness:
                 // all indices in `indices` are by construction `< array.len() == values.len()`
                 // limit is by construction < indices.len()
-                let limit = limit - validity.null_count();
+                let limit = limit.saturating_sub(validity.null_count());
```

Comment:
This is a bug fix I believe. could we add a test for it?

Reply:
We already checked `limit  > validity.null_count()`, so it may not be a bug, but better to add this line.